pkg/arvo/sur/eth-provider.hoon

```diff
@@ -0,0 +1,85 @@
+/-  json-rpc
+/+  ethereum
+=,  ethereum-types
+=,  jael
+|%
++$  url  @ta
+
++$  local
+  $:  =url
+  ==
++$  provider
+  $:  =url
+      kids=?  :: allow stars to receive requests from their planets
+      clients=(set @p)
+  ==
++$  client
+  $:  provider=@p
+  ==
++$  active  ?(%local %provider %client)
+
++$  state
+  $:  =active
+      =local
+      =provider
+      =client
+  ==
++$  topics  (list ?(@ux (list @ux)))
+
++$  action
+  $%  [%set-local =local]
+      [%set-provider =provider]
+      [%set-client =client]
+      [%provide tid=@ta =ethin]
+      [%set-kids kids=?]
+      [%add-client client=@p]
+      [%remove-client client=@p]
+  ==
+:: +$  update
+::   $%  [%get-state =mode]
+::   ==
+
+:: for starting a thread
++$  ethin
+  $%  
```

Comment:
Now the eth-provider is a drop-in replacement to ethio. The ethio requests in threads are basically forwarded as they are to the provider, but if we go with the `$request:rpc` type we would need to convert the `ethio` requests to `$request:rpc` and back. I think it might make sense to use `$request:rpc` if people would want to send requests not in the `ethio` format, but otherwise it is unnecessary to change this.